controllers/hyperparametertuningjob/hyperparametertuningjob_controller_test.go

```diff
@@ -1701,34 +661,153 @@ func createHpoJob(withFinalizer bool, k8sName, k8sNamespace, sageMakerName, regi
+	var actual hpojobv1.HyperparameterTuningJob
+	err := k8sClient.Get(context.Background(), types.NamespacedName{
+		Namespace: tuningJob.ObjectMeta.Namespace,
+		Name:      tuningJob.ObjectMeta.Name,
+	}, &actual)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(*actual.Status.TrainingJobStatusCounters.Completed).To(Equal(*counters.Completed))
+}
+
+// Helper function to create a DescribeHyperParameterTuningJobOutput.
+func CreateDescribeOutputWithOnlyStatus(status sagemaker.HyperParameterTuningJobStatus, bestTrainingJob *sagemaker.HyperParameterTrainingJobSummary, statusCounters *sagemaker.TrainingJobStatusCounters) sagemaker.DescribeHyperParameterTuningJobOutput {
+	return sagemaker.DescribeHyperParameterTuningJobOutput{
 		HyperParameterTuningJobStatus: status,
+		BestTrainingJob:               bestTrainingJob,
+		TrainingJobStatusCounters:     statusCounters,
 	}
-	err := k8sClient.Status().Update(context.Background(), job)
-	Expect(err).ToNot(HaveOccurred())
+}
+
+// Helper function to verify that there was an attempt to create a HyperParameterTuningJob with a given name.
+func ExpectRequestToCreateHyperParameterTuningJob(req interface{}, tuningJobName string) {
+	Expect(req).To(BeAssignableToTypeOf((*sagemaker.CreateHyperParameterTuningJobInput)(nil)))
+
+	createdRequest := req.(*sagemaker.CreateHyperParameterTuningJobInput)
+	Expect(*createdRequest.HyperParameterTuningJobName).To(Equal(tuningJobName))
+}
+
+// Helper function to verify that the specified object is a StopHyperParameterTuningJobInput and that it requests to delete the HyperParameterTuningJob.
+func ExpectRequestToStopHyperParameterTuningJob(req interface{}, tuningJob *hpojobv1.HyperparameterTuningJob) {
+	Expect(req).To(BeAssignableToTypeOf((*sagemaker.StopHyperParameterTuningJobInput)(nil)))
+
+	stopRequest := req.(*sagemaker.StopHyperParameterTuningJobInput)
+	Expect(*stopRequest.HyperParameterTuningJobName).To(Equal(controllers.GetGeneratedJobName(tuningJob.ObjectMeta.GetUID(), tuningJob.ObjectMeta.GetName(), MaxHyperParameterTuningJobNameLength)))
+}
+
+// Helper function to verify that the controller attempted to delete the spawned training jobs.
+func ExpectDeletedSpawnedTrainingJobs(spawner mockTrackingHPOTrainingJobSpawner, calls int, job hpojobv1.HyperparameterTuningJob) {
+	Expect(spawner.deleteSpawnedTrainingJobsCalls.Len()).To(Equal(calls))
+
+	calledJob := spawner.spawnMissingTrainingJobsCalls.Front().Value.(hpojobv1.HyperparameterTuningJob)
+	Expect(*calledJob.Spec.Region).To(Equal(*job.Spec.Region))
+	Expect(calledJob.ObjectMeta.GetName()).To(Equal(job.ObjectMeta.GetName()))
+	Expect(calledJob.ObjectMeta.GetNamespace()).To(Equal(job.ObjectMeta.GetNamespace()))
+}
+
+// Helper function to verify that the controller attempted to spawn the missing child training jobs.
+func ExpectSpawnMissingTrainingJobs(spawner mockTrackingHPOTrainingJobSpawner, calls int, job hpojobv1.HyperparameterTuningJob) {
```

Comment:
The naming choices here make usage of this function confusing. These questions came to my mind when encountering this in one of the tests: What does `1` mean? Does this assert all calls for the function, or just one? Is it testing the call parameters, or just that it was called?

We should either have it specifically look at the first call, or take a call index (_a la_ [ExpectNthSubreconcilerCallToKeepUnusedResources](https://github.com/aws/amazon-sagemaker-operator-for-k8s/blob/master/controllers/hosting/hostingdeployment_controller_test.go#L1036)). We should also pick a function name that better fits what it does, maybe something like `ExpectNthSpawnMissingTrainingJobsCallToHaveHPOJob`? I'm open to better names as long as they convey the purpose reasonably well.

Same goes for `ExpectDeletedSpawnedTrainingJobs`.